### ISSUE TITLE
Recipe #261: Annotation with a Non-Rectangular Polygon 

### ISSUE THREAD

**Issue by glenrobson** [IIIF/trc#80]
## Links

 * Original Issue: https://github.com/IIIF/cookbook-recipes/issues/261
 * Pull Request: https://github.com/IIIF/cookbook-recipes/pull/262
 * Preview: https://preview.iiif.io/cookbook/0261-nonrectangular-commenting/recipe/0261-non-rectangular-commenting/

## Background and Summary

Building on the successful approval of #78 and #77 this recipe demonstrates how to make an annotation that targets a non rectangular area using a SVG shape as the target. This recipe is supported in Mirador 3.

## Voting and changes

We welcome comments on the recipe and as well as voting +1, confused face or -1 feel free to add comments to this issue. If this issue is approved then the author will take account of the comments before we merge the branch in to the master cookbook branch.

If the recipe is rejected by the TRC then we will make the changes requested and resubmit it to a future TRC meeting. If you feel that your comments are substantial enough that the recipe should be looked at again by the TRC after the changes have been made please vote -1 (thumbs down). A confused face is treated as abstaining. 

Changes to the recipe will only be made after the TRC voting process has concluded.

**Comment by julsraemy** [comment]
Just a small remark: in the first paragraph of the `Implementation Notes`, the hyperlink to the Presentation API 3.0 is missing (I guess it's just an omission in Markdown).

**Comment by kirschbombe** [comment]
Should 'manifest' in the final paragraph of the Implementation Notes be capitalized?

**Comment by triplingual** [comment]
Entering my routine note that I voted 👍🏻 for my recipe

**Comment by azaroth42** [comment]
:-1: Strongly feel that recipes for 3.0 should not refer to 2.1 but instead either describe what is needed, or point to the relevant specification. (Also various editorial issues as others have commented)

**Comment by glenrobson** [comment]
@azaroth42 would changing the link from 2.1 to https://www.w3.org/TR/annotation-model/#svg-selector change your vote? It talks briefly about the scale of the SVG being relative to the image so we might also need to take some of the canvas language from 2.1 into the recipe.

**Comment by zimeon** [comment]
Like @azaroth42, I feel strongly that we should not link into the 2.1 spec. I would be OK copy or paraphrase content from 2.1, but link only to 3! Hence 👎   vote

**Comment by azaroth42** [comment]
Yes, either referring to another current spec (and svg-selector is the obvious candidate) or just importing the content from 2.1 into the recipe, which is probably better as it's more specific guidance for this particular scenario. Indeed, the notion of the cookbook came from discussions about how to avoid the content in the 2.1 spec about uses of annotations at the beginning of the work on 3.0 -- that instead we should refer to other specs for definitions, and have recipes / implementation notes to describe how to implement particular use cases. So it's great that the recipe is bringing that to fruition ... it should just not do it by pointing to the 2.x spec as if 3.x was mistakenly missing something.

**Comment by triplingual** [comment]
Noting with bemusement that an earlier version of this recipe quoted the relevant part of 2.1 inline. 😇

**Comment by azaroth42** [comment]
:+1: to the earlier version :)

**Comment by mixterj** [comment]
Along with @zimeon and @azaroth42 I think we should not link to 2.1 spec. I also think that in-line text is the best option.

**Comment by glenrobson** [comment]
### Issue 80 (Recipe #261: Annotation with a Non-Rectangular Polygon )
  +1: 13 [awead cubap glenrobson irv jcreel kirschbombe markpatton markpbaggett mbennett-uoe nfreire shuddles thehabes triplingual]
   0: 0 []
  -1: 11 [azaroth42 dlpierce jonhartzler jtweed julsraemy ksclarke mcwhitaker mixterj rsinghal tpendragon zimeon]
  Not TRC: 0 []
  Ineligible: 1 [scossu]

### Result: 13 / 24 = 0.54
No super majority, issue is referred to ex officio for decision

**Comment by regisrob** [comment]
While reading this recipe I wonder if it would make sense to add a note on the fact that when using non-rectangular annotations it is sometimes useful to also capture the rectangular region that comprises the polygon, so that it can be easily extracted and displayed using the Image API. 
But it should rather be a separate recipe explaining this specific use case, i.e. when you want to use non-rectangular shapes to annotate an image and be able to display the annotated image regions in a web page (which would imply to use a `Choice` of selectors if I recall correctly...). Is there already a candidate recipe for that? If not, does it make sense to have one?

**Comment by triplingual** [comment]
@regisrob That sounds like a good candidate for a recipe, taking this recipe (how to include a non-rectangular annotation) to an additional level of utility and explanation (one part of consequences for using a non-rectangular polygon annotation). We thought among the cookbook authors that this imagined recipe can be mentioned and speculatively linked to in this one.

**Comment by regisrob** [comment]
@triplingual I agree, and I'd be interested in writing this potential candidate recipe, but the main problem is the time available (at least until mid-December...)

**Comment by glenrobson** [comment]
The requested changes have been made here:

https://github.com/IIIF/cookbook-recipes/pull/262/commits/89d614618f6b0d18e070bfe2645d78a49e4250fb

Please could the TRC ex-officio members vote on this comment either thumbs up/down or confused face to indicate your acceptance now the changes have been made?

**Comment by glenrobson** [comment]
Comment from the ex-offico's:

"I think the “(noticable if the 0,0 point is different than the Canvas’s)” addition should be deleted because it isn’t always correct — think of two rectangles both anchored at 0,0 but different"

**Comment by glenrobson** [comment]
Ex-officos have approved the changes Trip has made so I will merge the recipe.
